src/wallet/walletdb.cpp

```diff
@@ -471,6 +483,19 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
             // so set the wallet birthday to the beginning of time.
             pwallet->nTimeFirstKey = 1;
         }
+        else if (strType == "vkey")
```

Comment:
``vkey`` needs to be added to the ``IsKeyType`` function, I think.

Reply:
Right, but viewing keys should also be so protected.

Reply:
This is resolved now.